pydicom/filereader.py

```diff
@@ -912,28 +912,57 @@ def data_element_offset_to_value(is_implicit_VR, VR):
     return offset
 
 
-def read_deferred_data_element(fileobj_type, filename, timestamp,
+def read_deferred_data_element(fileobj_type, filename_or_obj, timestamp,
                                raw_data_elem):
     """Read the previously deferred value from the file into memory
-    and return a raw data element"""
+    and return a raw data element.
+
+    Parameters
+    ----------
+    fileobj_type : type
+        The type of the original file object.
+    filename_or_obj : str or file-like
+        The filename of the original file if one exists, or the file-like
+        object where the data element persists.
+    timestamp : time or None
+        The time the original file has been read, if not a file-like.
+    raw_data_elem : dataelem.RawDataElement
+        The raw data element with no value set.
+
+    Returns
+    -------
+    dataelem.RawDataElement
+        The data element with the value set.
+
+    Raises
+    ------
+    IOError
+        If `filename_or_obj` is ``None``.
+    IOError
+        If `filename_or_obj` is a filename and the corresponding file does
+        not exist.
+    ValueError
+        If the VR or tag of `raw_data_elem` does not match the read value.
+    """
     logger.debug("Reading deferred element %r" % str(raw_data_elem.tag))
     # If it wasn't read from a file, then return an error
-    if filename is None:
+    if filename_or_obj is None:
         raise IOError("Deferred read -- original filename not stored. "
                       "Cannot re-open")
+    is_file = isinstance(filename_or_obj, compat.string_types)
+
```

Comment:
`is_filename` would be more descriptive here - when I first saw the name I expected a check for a file object.  It also makes it parallel to `filename_or_obj`.

Reply:
Ok, done.